File: conformance/bundles/11-emoji-offsets/greet.steps.py
from varar import steps

param, stimulus, sensor = steps(lambda: {})


# Two slots: the {string} capture and the trailing (non-header-bound) table,
# which arrives in the splat. Both are echoed back so the core actually
# compares them — the table's data rows only, since the header row is labels.
@sensor("I greet {string}")
def _(state, s, *extra):
    return [s, extra[0][1:]]

File: conformance/bundles/11-emoji-offsets/greet.steps.cs
// C# sibling of greet.steps.ts / .rs (bundle 11-emoji-offsets).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B11;

public static class GreetSteps
{
    public static void Register(Steps s)
    {
        // The list item is followed by a table, appended as a trailing arg, so this sensor has two
        // slots: {string} and the table. Both are echoed back so the core compares them — the
        // table's data rows only, since the header row is labels and is never compared.
        s.Sensor("I greet {string}", (state, name, table) =>
        {
            var rows = ((VList)table).Items.Skip(1).ToList();
            return Value.List([name, Value.List(rows)]);
        });
    }

    public static Value State() => Value.Null;
}
